test/controllers/users_controller_test.rb

```diff
@@ -90,6 +90,18 @@ def test_show
     assert_select(
       "a[href = '#{name_descriptions_path}?by_author=#{user.id}']"
     )
+    assert_select(
+      "a:match('href', ?)",
+      /\?.*=(\d+).*&id=\1/, # some param=n followed by id with same value
+      false,
+      "Links should not use the same value for id and another param"
+    )
+    assert_select(
+      "a:match('href', ?)",
+      /\?.*id=(\d+).*&\S+=\1/, # id=n followed by another param with same value
+      false,
+      "Links should not use the same value for id and another param"
+    )
```

Comment:
Is there a way to combine the regexp's (in order to use a single `assert_select`)/

Reply:
Only think i can imagine is if you just omit the `id` and search for two `=(\d+)`. 
Doesn't seem safe though, what if one of the params has very few digits?